app/controllers/category_transactions_controller.rb

```diff
@@ -0,0 +1,32 @@
+class CategoryTransactionsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_category, only: :new
+
+  def new
+  end
+
+  def create
+    @category_transaction = CategoryTransactions::Creator.new(category_transaction_params)
+    respond_to do |format|
+      if @category_transaction.call
+        flash[:notice] = t('transaction_create')
+      else
+        flash[:alert] = t('category_transaction_not_create')
+      end
+      format.html { redirect_to categories_path }
+      format.js
+    end
+  end
```

Comment:
@marderer let's give more space for `#create` method while moving `@category_transaction` out of it, see:
```ruby
def create
  if category_transaction_creator.create
    flash[:notice] = t('transaction_create')
  else
    flash[:alert] = t('category_transaction_not_create')
  end
  
  respond_to do |format|
    format.html { redirect_to categories_path }
    format.js
  end
end

def category_transaction_creator
  CategoryTransactions::Creator.new(category_transaction_params)
end
```